pkg/apis/componentconfig/types.go

```diff
@@ -176,6 +176,14 @@ const (
 type KubeletConfiguration struct {
 	metav1.TypeMeta
 
+	// Only used for dynamic configuration.
+	// The length of the trial period for this configuration. Default 10 minutes.
+	ConfigTrialDuration metav1.Duration
+	// Only used for dynamic configuration.
+	// The maximum number of Kubelet restarts that can occur in ConfigTrialDuration before the
+	// configuration fails the trial and the Kubelet rolls back to its last-known-good config.
+	// Default 3, mimimum allowed is 2, maximum allowed is 10.
```

Comment:
I made it so the minimum can be 1. I'd like to postpone trying to differentiate crashes from config-exits, as I have a feeling any hasty solution I dream up now will be riddled with edge-cases.